parser.py

```diff
@@ -422,18 +434,23 @@ def course_attrs(self):
                 if box_title == COURSE_DETAIL:
                     # Units and course components
                     labels = table.find_all("span", {"class": EDITBOX_LABEL_CLASS})
+                    
                     data = table.find_all("span", {"class": EDITBOX_DATA_CLASS})
+
+                    dx = 0
```

Comment:
I tried to explain it a bit better and changed the name to dataIndex. It mirrors x until the course components happen, then it gets fast forwarded to the index after the last i+1, then it continues to advance by one each time x increments